wavenet.py

```diff
@@ -106,22 +106,24 @@ def _create_dilation_layer(self,
 
     def encode(self, audio):
         '''Quantizes waveform amplitudes.'''
-        with tf.name_scope('preprocessing'):
+        with tf.name_scope('encode'):
             mu = self.quantization_channels - 1
             # Perform mu-law companding transformation (ITU-T, 1988).
             magnitude = tf.log(1 + mu * tf.abs(audio)) / tf.log(1. + mu)
             signal = tf.sign(audio) * magnitude
-            # Quantize signal to the specified number of levels.
-            quantized = tf.cast((signal + 1) / 2 * mu, tf.int32)
-        return quantized
+            # Quantize signal to the specified number of levels
+            return tf.cast((signal + 1) / 2 * mu + 0.5, tf.int32)
 
 
     def decode(self, output):
-        mu = self.quantization_channels - 1
-        y = tf.cast(output, tf.float32)
-        y = 2 * (y / mu) - 1
-        x = tf.sign(y) * (1 / mu) * ((1 + mu)**abs(y) - 1)
-        return x
+        '''Recovers waveform from quantized values.'''
+        with tf.name_scope('decode'):
+            mu = self.quantization_channels - 1
+            # Map values back to [-1, 1]
+            casted = tf.cast(output, tf.float32)
+            y = 2 * (casted / mu) - 1
+            # Perform inverse of mu-law transformation
+            return tf.sign(y) * (1 / mu) * ((1 + mu)**abs(y) - 1)
```

Comment:
Yes, I understand putting no space around `**` is reasonable.
In terms of PEP8, it force putting space on some operators, but not for `**` (and I've just ran pep8) , so it's okay 👍

Let's go for it.